pkg/executor/build.go

```diff
@@ -587,16 +586,28 @@ func DoBuild(opts *config.KanikoOptions) (v1.Image, error) {
 	return nil, err
 }
 
+// fileToSave returns all the files matching the given pattern in deps.
+// If a file is a symlink, it also returns the target file.
+// It first returns all the target files and then symlinks so they can copied
+// in that order to the kaniko workspace.
```

Comment:
Is there a requirement or benefit to copying them in order?

Reply:
Previously, i was using `filepath.EvalSymlink` and it was requirement then or else `EvalSymlink` returned an error "target file does not exist"
With the current implementation it is no longer a requirement
```
for _, p := range filesToSave {
			logrus.Infof("Saving file %s for later use", p)
			util.CopyFileOrSymlink(p, dstDir)
		}

----
// For cross stage dependencies kaniko must persist the referenced path so that it can be used in
// the dependent stage. For symlinks we copy the target path because copying the symlink would
// result in a dead link
func CopyFileOrSymlink(src string, destDir string) error {
	destFile := filepath.Join(destDir, src)
	if fi, _ := os.Lstat(src); IsSymlink(fi) {
		link, err := os.Readlink(src)
		if err != nil {
			return err
		}
		return os.Symlink(link, destFile)
	}
	return otiai10Cpy.Copy(src, destFile)
}
```

Reply:
Let me try not retaining the order and see if it still matters.